feat: add contexts and types for lists and tasks

Aadarsh-shrivastava opened the pull request:
add contexts and type for lists and tasks data management

Aadarsh-shrivastava committed:
feat: add contexts and types for lists and tasks

shah-harshit commented on src/types/List.ts:
```diff
@@ -0,0 +1,11 @@
+import { Task } from "./Task";
+
+export type List = {
+    id:number;
```
Sort and use prettier for formatting.

shah-harshit commented on src/types/Task.ts:
```diff
@@ -0,0 +1,10 @@
+export type Task = {
+    id:number;
+    name:string;
+    isDone:boolean;
```
Sort and use prettier for formatting.

shah-harshit commented on src/contexts/ListsContext.tsx:
```diff
@@ -0,0 +1,44 @@
+
+interface ListsContextProps {
+  lists: List[];
+  addList: (list: List) => void;
```
Sort everywhere and use prettier for formatting.

shah-harshit commented on src/contexts/TasksContext.tsx:
```diff
@@ -0,0 +1,48 @@
+interface TasksContextProps {
+  Tasks: Task[];
+  addTask: (Task: Task) => void;
+  deleteTask: (id: number) => void;
```
Sort everywhere and use prettier for formatting.

shah-harshit commented on src/contexts/TasksContext.tsx:
```diff
@@ -0,0 +1,48 @@
+
+  const updateTask = (id: number, updatedTask: Partial<Task>) => {
+    setTasks((prevTasks) =>
+      prevTasks.map((Task) =>
```
```suggestion
      prevTasks.map((task) =>
```

shah-harshit reviewed (commented):
As we discussed, we should use a single context for both of them, since this is a small, single-page application with six combined operations.

Aadarsh-shrivastava committed:
refactor: sort and format code properly
chore: change eslint config for better linting
update: use single context for lists and tasks

Aadarsh-shrivastava committed:
fix: eslint config

Aadarsh-shrivastava committed:
refactor: sort list context props interface
